src/vcpkg/postbuildlint.cpp

```diff
@@ -578,6 +581,26 @@ namespace vcpkg::PostBuildLint
             }
         }
 
+        if (!binaries_with_invalid_architecture.empty())
+        {
+            print_invalid_architecture_files(expected_architecture, binaries_with_invalid_architecture);
+            return LintStatus::ERROR_DETECTED;
+        }
+#elif defined(__APPLE__)
+        std::vector<FileAndArch> binaries_with_invalid_architecture;
+        const auto requested_arch = expected_architecture == "x64" ? "x86_64" : expected_architecture;
+        for (const Path& file : files)
+        {
+            auto cmd_line = Command("lipo").string_arg("-archs").path_arg(file);
+            ExitCodeAndOutput ec_data = cmd_execute_and_capture_output(cmd_line);
+            Checks::check_exit(
+                VCPKG_LINE_INFO, ec_data.exit_code == 0, "Running command:\n   %s\n failed", cmd_line.command_line());
+            if (ec_data.output.find(requested_arch) == std::string::npos)
```

Comment:
Will this find `x86` inside `x86_64`?

Reply:
There is no `x86` on macOS. They removed the 32 bit support on macOS and ios a few years ago.

Reply:
it was like 10.7 where they removed x86_32 machine support, and 10.14 is the last version of macOS that supported 32-bit applications. However, I'd still appreciate doing a split on spaces and checking if any of the architectures equal, not just doing a find.